c_glib/plasma-glib/object.cpp

```diff
@@ -0,0 +1,537 @@
+                          "GPU device",
+                          "The GPU device number. -1 means GPU isn't used.",
+                          -1,
+                          G_MAXINT,
+                          -1,
+                          static_cast<GParamFlags>(G_PARAM_READWRITE |
+                                                   G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_GPU_DEVICE, spec);
+}
+
+static bool
+gplasma_object_check_not_released(GPlasmaObjectPrivate *priv,
+                                  GError **error,
+                                  const gchar *context)
+{
+  if (priv->client) {
+    return true;
+  }
+
+  auto id_priv = GPLASMA_OBJECT_ID_GET_PRIVATE(priv->id);
+  auto id_hex = id_priv->id.hex();
+  g_set_error(error,
+              GARROW_ERROR,
+              GARROW_ERROR_INVALID,
+              "%s: Can't process released object: <%s>",
+              context,
+              id_hex.c_str());
+  return false;
+}
+
+static void
+gplasma_object_release_resources(GPlasmaObjectPrivate *priv)
+{
+  if (priv->client) {
+    g_object_unref(priv->client);
+    priv->client = nullptr;
+  }
+
+  if (priv->data) {
+    g_object_unref(priv->data);
+    priv->data = nullptr;
+  }
+
+  if (priv->metadata) {
+    g_object_unref(priv->metadata);
+    priv->metadata = nullptr;
+  }
+}
+
+G_DEFINE_TYPE(GPlasmaCreatedObject,
+              gplasma_created_object,
+              GPLASMA_TYPE_OBJECT)
+
+static void
+gplasma_created_object_dispose(GObject *object)
+{
+  auto priv = GPLASMA_OBJECT_GET_PRIVATE(object);
+
+  if (priv->client) {
+    gplasma_created_object_abort(GPLASMA_CREATED_OBJECT(object), NULL);
+  }
+
+  G_OBJECT_CLASS(gplasma_created_object_parent_class)->dispose(object);
+}
+
+static void
+gplasma_created_object_init(GPlasmaCreatedObject *object)
+{
+}
+
+static void
+gplasma_created_object_class_init(GPlasmaCreatedObjectClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->dispose = gplasma_created_object_dispose;
+}
+
+/**
+ * gplasma_created_object_seal:
+ * @object: A #GPlasmaCreatedObject.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Seals the object in the object store. You can't use the sealed
+ * object anymore.
+ *
+ * Returns: %TRUE on success, %FALSE on error.
+ *
+ * Since: 0.12.0
+ */
+gboolean
+gplasma_created_object_seal(GPlasmaCreatedObject *object,
+                            GError **error)
+{
+  const auto context = "[plasma][created-object][seal]";
+
+  auto priv = GPLASMA_OBJECT_GET_PRIVATE(object);
+  if (!gplasma_object_check_not_released(priv, error, context)) {
+    return FALSE;
+  }
+
+  auto plasma_client = gplasma_client_get_raw(priv->client);
+  auto id_priv = GPLASMA_OBJECT_ID_GET_PRIVATE(priv->id);
+  auto status = plasma_client->Seal(id_priv->id);
+  auto success = garrow_error_check(error, status, context);
+  if (success) {
+    plasma_client->Release(id_priv->id);
```

Comment:
Do we need to receive `status` to silence the warning?
https://travis-ci.org/apache/arrow/jobs/461631864#L3940

Reply:
Yes.
I've added it.